test/test_autograd.py

```diff
@@ -3460,6 +3460,68 @@ def backward(ctx, gO):
                     out.backward()
             self.assertIn('MyFunc.apply', str(w[0].message))
 
+    def test_nested_anomaly_detect_nan(self):
+        size = 10
+
+        class MyFunc(Function):
+            @staticmethod
+            def forward(ctx, inp1, fail_0th):
+                ctx.fail_0th = fail_0th
```

Comment:
You should not save input/output directly on the ctx. This will lead to a memory leak.
You should do `ctx.save_for_backward(inp1, fail_0th)` and recover them in the backward with `inp1, fail_0th = ctx.saved_tensors`.

Same for the other custom Functions below.

Reply:
Ah! I've been wondering the advantage of using `save_for_backward` over saving it directly to `ctx`. Is it on the documentation?

Reply:
I got this error when I did this:

    Traceback (most recent call last):
      File "pytorch/test/test_autograd.py", line 3500, in test_nested_anomaly_detect_nan
        out = MyFunc.apply(inp, True)
    TypeError: save_for_backward can only save variables, but argument 1 is of type bool

Reply:
Ho fail_0th is not a Tensor sorry. You can save it in the ctx then :D Only the Tensor input/output have to follow this.

You can check the doc on how to write a custom Function for more details here: https://pytorch.org/docs/stable/notes/extending.html